modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -286,6 +294,39 @@ func (p *ProjectPipelineService) Create(ctx context.Context, params *pb.CreatePr
 	}}, nil
 }
 
+func (p *ProjectPipelineService) createCronIfNotExist(definition *dpb.PipelineDefinition, projectPipelineType ProjectSourceType) error {
+	extraString := projectPipelineType.GetPipelineCreateRequestV2()
+	var extra apistructs.PipelineDefinitionExtraValue
+	err := json.Unmarshal([]byte(extraString), &extra)
+	if err != nil {
+		return err
+	}
+
+	crons, err := p.PipelineCron.CronPaging(context.Background(), &cronpb.CronPagingRequest{
```

Comment:
I think you should also search existed-cron by definition id.

Reply:
This method is called when the definition is created, there can not be have cron that contains the definition just created